src/daft-schema/src/python/datatype.rs

```diff
@@ -360,6 +361,10 @@ impl PyDataType {
         Ok(self.dtype.is_numeric())
     }
 
+    pub fn is_float(&self) -> PyResult<bool> {
+        Ok(self.dtype.is_floating())
+    }
```

Comment:
Are we able to now remove all of these methods along with all of the stubs in .pyi?

Reply:
Initially, I thought we'd be able to remove these, but i actually had to add a bunch more as we roundtrip from py<-->rust, so i needed to add an `is_T` for every variant, along with a `get_T_inner` for every variant that contains inner values.

Reply:
so I think considering I needed to add all of these anyways, there _could_ be a compelling argument for just exposing these to `DataType` and not do subclassing 🤷

Reply:
Ok, I agree that this is getting a bit convoluted and have all fields with attribute errors when missing is no biggie. At the end of the day, the customer ask is to be able to check if something is a timestamp type. If you want/need all of the `is_` methods, then that's cool too but perhaps kept internal. One issue with classes is we have to create our entire type lattice with a python class hierarchy which is a pain and requires introducing a ton of subclasses. We also have missing parts of the lattice at the moment like "is_exact_numeric" which requires an additional layer of subclasses.